### PR TITLE
[Add] Load option for user config

### PR DESCRIPTION
**Pull request by alusor**

### CODE REVIEW

**Comment by isaachinman** [comment]
Hi @alusor, thanks for the PR.

First - can you please remove the `package-lock.json` files? This project uses `yarn`.

Second - this is a good idea, but in general we need to load all namespaces for SSR to work correctly. Did you have a specific use case? Can you explain your thoughts? Perhaps we want to load all namespaces on the server, but pass this user-defined `load` option to the clientside instance only?

**Comment by alusor** [comment]
Sure i will remove it.
Second - Yes i think we need to load all namespaces, the thing is that in your source, the load have 'languageOnly' value, in my case i have 'es' and 'es-CO' so i need to set the value to 'all' to correctly render 'es-CO'

**Comment by isaachinman** [comment]
Oh sorry, I misunderstood. This is the `load` option from `i18next` directly that essentially enables or disables sub-language locales. The two main options we'd want to allow here would be `languageOnly` and `all`.

Once this PR is cleaned up a bit I'll be happy to accept it.

File: src/config/create-config.js [hunk 1]
```diff
@@ -10,19 +10,23 @@ export default (userConfig) => {
 
   combinedConfig.allLanguages = combinedConfig.otherLanguages
     .concat([combinedConfig.defaultLanguage])
+  combinedConfig.load = combinedConfig.load || 'languageOnly'
```
Comment (isaachinman) [inline src/config/create-config.js]: Default values should go inside `default-config.js`, not via the pipe operator like this.

File: src/config/create-config.js [hunk 2]
```diff
@@ -10,19 +10,23 @@ export default (userConfig) => {
...
+
 
   if (isNode && !process.browser) {
     const fs = eval("require('fs')") // eslint-disable-line
     const path = require('path')
 
     const getAllNamespaces = p => fs.readdirSync(p).map(file => file.replace('.json', ''))
     const {
-      allLanguages, defaultLanguage, localePath, localeStructure,
+      allLanguages, defaultLanguage, localePath, localeStructure, load, fallbackLng
```
Comment (isaachinman) [inline src/config/create-config.js]: Please keep this alphabetical.

**Comment by isaachinman** [comment]
Ah, and this new option will need to be added to the README.

**Comment by isaachinman** [comment]
@alusor Do you know if you'll have time to tidy this up? Otherwise no worries, I will take care of it.

**Comment by alusor** [comment]
done, i was do the necessary changes

**Comment by kachkaev** [comment]
There is still package-lock.json in the example folder :)

**Comment by isaachinman** [comment]
This PR is actually unnecessary. @alusor Thank you for your time regardless.

You can pass `load` into the class constructor:

```jsx
new NextI18Next({ load: 'all' })
```

**Pull request closed without merging**